### PR TITLE
Make internal task unique per papermill task

### PR DESCRIPTION
**Pull request by EngHabu**
Signed-off-by: Haytham Abuelfutuh <haytham@afutuh.com>

# TL;DR
Papermill task plugin creates a task with a dummy function. Because it uses the same function definition for every instance of the NotebookTask, that results into a naming conflict. This PR uses the name of the Notebook task when creating the python task to guarantee uniqueness (or at least as unique as Notebook tasks)

## Type
 - [X] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [X] Code completed
 - [X] Smoke tested
 - [X] Unit tests added
 - [X] Code documentation added
 - [X] Any pending items have an associated Issue

## Tracking Issue
https://github.com/flyteorg/flyte/issues/1460

### CODE REVIEW

**Review by wild-endeavor — DISMISSED** [review]

File: plugins/flytekit-papermill/flytekitplugins/papermill/task.py [hunk 1]
```diff
@@ -111,6 +111,8 @@ def __init__(
         outputs: typing.Optional[typing.Dict[str, typing.Type]] = None,
         **kwargs,
     ):
+        _dummy_task_func.__name__ = name
```
Comment (eapolinario) [inline plugins/flytekit-papermill/flytekitplugins/papermill/task.py:114]: Can you leave comment and explain why we're doing this?

**Review by wild-endeavor — APPROVED** [review]
